packit_service/service/models.py

```diff
@@ -0,0 +1,87 @@
+"""
+Data models for jobs, tasks, builds, etc.
+"""
+import datetime
+from typing import List, Dict
+
+from frambo.dict_in_redis import PersistentDict
+
+
+class Model:
+    table_name: str
+    identifier: str
+
+    @classmethod
+    def db(cls):
+        if not cls.table_name:
+            raise RuntimeError("table_name is not set")
+        return PersistentDict(hash_name=cls.table_name)
+
+    def save(self):
+        if not self.identifier:
+            raise RuntimeError("identifier not set")
+        self.db()[self.identifier] = self.__dict__
+
+    def __str__(self):
+        return f"{self.table_name} - {self.__dict__}"
+
+
+class User(Model):
+    """ A GitHub user """
+
+    table_name = "users"
+    username: str
+
+
+class Installation:
```

Comment:
I know that it is still WIP, but why does `Installation` not inherit from `Model`? And could we use here `events.py` classes somehow?

Reply:
I worked on this before you opened yours, definitely planning on using your work and change this code.

I'm also planning to scope this down and only do tasks, builds and installs